sklearn/metrics/cluster/unsupervised.py

```diff
@@ -347,6 +351,7 @@ def davies_bouldin_score(X, labels):
     if np.allclose(intra_dists, 0) or np.allclose(centroid_distances, 0):
         return 0.0
 
-    score = (intra_dists[:, None] + intra_dists) / centroid_distances
-    score[score == np.inf] = np.nan
-    return np.mean(np.nanmax(score, axis=1))
+    centroid_distances[centroid_distances == 0] = np.inf
+    combined_intra_dists = intra_dists[:, None] + intra_dists
+    scores = np.amax(combined_intra_dists / centroid_distances, axis=1)
```

Comment:
we are more familiar with the `np.max` alias

Reply:
Fixed in 5cb80be